src/mode/modeHandler.ts

```diff
@@ -147,6 +149,7 @@ export class SearchState {
      */
     public searchDirection = 1;
 
+    public vimSettings: VimSettings;
```

Comment:
We have `vimSettings` in `VimState` already and it's initialized in `modeHandler`. Besides, we have `searchState` in VimState, the model between VimState and SearchState is Composition. So what we are doing is accessing outter object's property from inner object, right? My current solution is passing `vimSettings` alone to `SearchState` while initalization as I don't want to any other part of `VimState`.

So are you saying we should pass VimState directly to SearchState while initializing SearchState?

Another way to handle this situation is moving `SearchState` definition into `VimState` as we don't use it without the outter VimState. Making it a subclass can provide us the ability to access members of VimState. Any idea?